src/buffers/disk/leveldb_buffer.rs

```diff
@@ -239,16 +247,30 @@ impl Reader {
 
             self.delete_offset = new_offset;
 
-            self.db.compact(&Key(0), &Key(self.delete_offset));
-
             let size_deleted = self.unacked_sizes.drain(..num_to_delete).sum();
             self.current_size.fetch_sub(size_deleted, Ordering::Relaxed);
+
+            self.uncompacted_size += size_deleted;
+            if self.uncompacted_size as f64
+                > self.max_size as f64 * MAX_UNCOMPACTED / (1.0 - MAX_UNCOMPACTED)
```

Comment:
It might be simpler to pass down a `max_uncompacted` the same way we pass down the new `max_size`, just to make this math easier to understand at a glance. Then it could be just `max_size * MAX_UNCOMPACTED` in `build` below. It's also clearer that we aren't recalculating the max every time.

Reply:
True, that's now much clearer.